libs/MeshKernel/include/MeshKernel/BoundingBox.hpp

```diff
@@ -0,0 +1,108 @@
+//---- GPL ---------------------------------------------------------------------
+//
+// Copyright (C)  Stichting Deltares, 2011-2021.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+// contact: delft3d.support@deltares.nl
+// Stichting Deltares
+// P.O. Box 177
+// 2600 MH Delft, The Netherlands
+//
+// All indications and logos of, and references to, "Delft3D" and "Deltares"
+// are registered trademarks of Stichting Deltares, and remain the property of
+// Stichting Deltares. All rights reserved.
+//
+//------------------------------------------------------------------------------
+
+#pragma once
+
+#include "Entities.hpp"
+
+#include <algorithm>
+
+namespace meshkernel
+{
+    class BoundingBox
+    {
+    public:
+        BoundingBox() = default;
+
+        BoundingBox(const Point& m_lower_left, const Point& m_upper_right)
+            : m_lowerLeft(m_lower_left),
+              m_upperRight(m_upper_right)
+        {
+        }

```

Comment:
Should we check that the lower left is really less than the upper right, otherwise throw an exception